S2VX.Game/Story/Note/GameNote.cs

```diff
@@ -52,40 +54,72 @@ public class GameNote : S2VXNote {
             Delete();
         }
 
-        // Notes are clickable if they are visible on screen and not missed
+        // Notes are clickable if they are visible on screen, not missed, and is the earliest note
         private bool IsClickable() {
             var time = Time.Current;
-            // Limit timing error to be +/- MissThreshold (though it will never be >= MissThreshold since RecordMiss would have already run) 
+            // Limit timing error to be +/- MissThreshold (though it will never be >= MissThreshold since RecordMiss would have already run)
             TimingError = (int)Math.Round(Math.Clamp(time - EndTime, -MissThreshold, MissThreshold));
-            return TimingError <= MissThreshold && Alpha > 0;
+            var inMissThreshold = TimingError <= MissThreshold && Alpha > 0;
+            var earliestNote = (GameNote)Story.Notes.Children[^1];
+            var isEarliestNote = earliestNote == this;
+            return inMissThreshold && isEarliestNote;
+        }
+
+        private void ClickNote() {
+            ScoreInfo.AddScore(Math.Abs(TimingError));
+            Delete();
+            BlockClicking = true;
+        }
+
+        // Unblocks clicking on the last call of OnMouseDown / OnKeyDown
+        private void CheckAndResetClicking() {
+            if (Story.Notes.Children.Count > 0 && this == Story.Notes.Children[0]) {
+                BlockClicking = false;
+            }
         }
 
         protected override bool OnMouseDown(MouseDownEvent e) {
-            if (IsClickable()) {
-                ScoreInfo.AddScore(Math.Abs(TimingError));
-                Delete();
+            if (IsClickable() && !BlockClicking) {
+                ClickNote();
             }
+
+            CheckAndResetClicking();
```

Comment:
For some reason even though it looks like the same logic, the reset blocking doesn't work for clicks

Reply:
I can confirm, clicks do not work. If you think about it, it does make sense, CheckAndResetClicking() only resets the BlockClicking if you are at the last note. No matter where your cursor is, KeyDown always triggers for all notes, but MouseDown may not trigger on the last note if your mouse is not over the note.